### ISSUE TITLE
Find a replacement for MarkdownSharp

### ISSUE THREAD

**Issue by riipah** [VocaDB/vocadb#38]
For server-side processing of Markdown VocaDB uses [MarkdownSharp](https://code.google.com/p/markdownsharp/), with some of its extra features turned on, roughly corresponding to the [Github-flavored Markdown](https://help.github.com/articles/github-flavored-markdown/) (GFM). Unfortunately the MarkdownSharp library hasn't been maintained in years and there's multiple libraries with much better performance available. Most of them don't support the same customizations that we're using though which is limiting our options to change. Switching to [CommonMark](http://commonmark.org/) could be an option as well once it's stabilized enough and there's a .NET parser for it.

List of extensions / deviations from the original Markdown spec:
- Automatic hyperlinks: no need to surround URLs with brackets (same as GFM)
- Strict bold italic: bolding and italics need non-word characters (such as whitespace or punctuation) on both sides (same as GFM).
- Automatic newlines (line breaks): every line break counts, no need to end the line with spaces or have multiple line breaks.

What we'd like to have additionally: option to disable images in comments (VocaDB doesn't really need these that often), or at least limit the maximum image size and file extension (to prevent animated gifs).

Some libraries to investigate:
- [MarkdownDeep](http://www.toptensoftware.com/markdowndeep/).
- [Moonshine](https://github.com/brandonc/moonshine).
- [CommonMark.NET](https://github.com/Knagis/CommonMark.NET/).

TL;DR: It'd be good if we could replace MarkdownSharp library with something more efficient, but we must be able to keep the current customizations.
